dot/rpc/subscription/websocket.go

```diff
@@ -265,14 +267,30 @@ func (c *WSConn) initAllBlocksListerner(reqID float64, _ interface{}) (Listener,
 }
 
 func (c *WSConn) initExtrinsicWatch(reqID float64, params interface{}) (Listener, error) {
-	pA := params.([]interface{})
+	var encodedExtrinsic string
 
-	if len(pA) != 1 {
-		return nil, errors.New("expecting only one parameter")
+	switch encodedHex := params.(type) {
+	case []string:
+		if len(encodedHex) != 1 {
+			return nil, fmt.Errorf("expected only 1 param item. got %d", len(encodedHex))
+		}
+		encodedExtrinsic = encodedHex[0]
+	case []interface{}:
+		if len(encodedHex) != 1 {
+			return nil, fmt.Errorf("expected only 1 param item. got %d", len(encodedHex))
+		}
+
+		var ok bool
+		encodedExtrinsic, ok = encodedHex[0].(string)
+		if !ok {
+			return nil, fmt.Errorf("expected type equals string. got %T", encodedHex[0])
+		}
+	default:
+		return nil, fmt.Errorf("expected type equals []string. got %T", params)
```

Comment:
use a sentinel error

also that should be `[]string or []interface{}` I guess? :thinking:

Reply:
The method `initExtrinsicWatch(reqID float64, params interface{})` expects a slice of 1 item that must be a string so there is the `case []interface{}` just to cover the case of `[]interface{"a"}` but the data should be always []string of 1 item

Reply:
Oh yes I was talking about the error string here

Reply:
I was talking about the error string. I made a code suggestion below 😉